src/assets/scripts/aircraft/AircraftController.js

```diff
@@ -33,45 +33,33 @@ export default class AircraftController {
 
     /**
      * @for AircraftController
-     * @method aircraft_generate_callsign
-     * @param airlineName
+     * @method aircraft_auto_toggle
      */
-    aircraft_generate_callsign(airlineName) {
-        // TODO: this should live in the AirlineModel
-        const airline = window.airlineController.airline_get(airlineName);
-
-        if (!airline) {
-            console.warn(`Airline not found: ${airlineName}`);
-
-            return `airline-${airlineName}-not-found`;
-        }
-
-        return airline.generateFlightNumber();
+    aircraft_auto_toggle() {
+        prop.aircraft.auto.enabled = !this.aircraft.auto.enabled;
     }
 
     /**
      * @for AircraftController
-     * @method aircraft_auto_toggle
+     * @method isCallsignInList
+     * @param callsign {string}
+     * return {boolean}
      */
-    aircraft_auto_toggle() {
-        prop.aircraft.auto.enabled = !this.aircraft.auto.enabled;
+    isCallsignInList(callsign) {
+        return this.aircraft.callsigns.indexOf(callsign) !== -1;
     }
 
     /**
      * @for AircraftController
-     * @method aircraft_callsign_new
-     * @param airline {string}
+     * @method addCallsignToList
+     * @param callsign {string}
      */
-    aircraft_callsign_new(airline) {
-        // TODO: the logic needs work here. if `callsign` is always initialized as null, one would imagine that
-        // this function would always result in the creation of a callsign?
-        let callsign = this.aircraft_generate_callsign(airline);
-
-        if (this.aircraft.callsigns.indexOf(callsign) === -1) {
-            this.aircraft.callsigns.push(callsign);
+    addCallsignToList(callsign) {
+        if (this.isCallsignInList(callsign)) {
+            return;
```

Comment:
It might be useful here to do a `console.warn` or something, as if we end up here with a callsign that's already in use, something has gone wrong, as it should be prevented before reaching the step of adding it to the callsign list.